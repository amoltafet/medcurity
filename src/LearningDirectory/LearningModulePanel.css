.LearningModulePanelCard {
    font-size: 3.8vmin;
    text-align: center;
    color: white;
    margin-top: .5%;
    margin-bottom: 6%;
    margin-left: 5%;
    margin-right: 0%;
    width: 40%;
    height: 25vmax;
    text-decoration: none;
    background-color:#1e5b88!important;
  
}

@media screen and (min-width: 1000px) {
    .LearningModulePanelCard {
        font-size: 1.8vmax;
        text-align: center;
        color: white;
        margin-top: .5%;
        margin-bottom: 3%;
        margin-right: 1.2%;
        margin-left: .6%;
        background-color:#1e5b88 !important;
        width: 18%;
        height: 20vmax;
        text-decoration: none;
    }
}

.LearningModulePanelCard:hover {
    background-color: #127ebf!important;
}