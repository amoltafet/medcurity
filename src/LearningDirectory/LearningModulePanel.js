import {Card} from 'react-bootstrap'
import React from 'react';
import './LearningModulePanel.css';
import { Button } from '@material-ui/core';


/**
 * Create a card object holding the learning module name and props
 * @param {props.id} The id of the module
 * @param {props.title} The title/name of the module
 * @returns Card object with the relevant information
 */
const LearningModulePanel = (props) => {
    var dueDate = new Date(props.dueDate); 
    return (
        <>
        <div className="card card-custom bg-white border-white border-0">
                <div className="card-custom-img card-custom-img-2"></div>
                    <div className="card-custom-avatar">
                        <img className="img-fluid" src={props.img} alt="Card image cap" />
                    </div>
                    <div className="card-body card-body-2">
                        <h4 className="card-title">{props.title}</h4>
                        <p className="card-text">Module {props.link}</p>
                        <h6 className="card-text">{dueDate}</h6>
                    </div>
                    <div className="card-footer card-footer-2" > 
                        <Button variant="outlined" href={"/learning-module/" + props.link}>View</Button>
                    </div>
                   
             </div>
        </>
    );
}
export default LearningModulePanel;