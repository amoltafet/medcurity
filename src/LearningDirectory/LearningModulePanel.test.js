import renderer from 'react-test-renderer';
import LearningModulePanel from './LearningModulePanel';
import React from 'react';

test('renders LoginPage', () => {
  const tree = renderer.create(<LearningModulePanel />).toJSON();
  expect(tree).toMatchSnapshot();
});