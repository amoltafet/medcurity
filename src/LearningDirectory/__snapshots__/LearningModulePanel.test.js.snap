// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LoginPage 1`] = `
<div
  className="LearningModulePanelCard uvs-right uvs-left card"
>
  <div
    className="card-body"
  >
    <a
      className="font card-link"
      href="/learning-module/undefined"
    />
  </div>
</div>
`;
